avm/src/daemon/daemon.rs

```diff
@@ -318,10 +318,15 @@ pub async fn start(is_local_anycloud_app: bool, local_agz_b64: Option<String>) {
           if control_port.is_leader() {
             // TODO: Should we keep these leader announcements in the stdout logging?
             println!("I am leader!");
-            match get_v1_stats().await {
-              Ok(s) => stats.push(s),
-              Err(err) => error!(NoStats, "{}", err).await,
-            };
+            // Do not collect stats until cluster is up. Otherwise, will have scaling issues.
+            if control_port.is_cluster_up() {
```

Comment:
Thinking about this after some sleep. There is a bit of a problem with this logic. It's true that during initial start or an upgrade some of the nodes won't be up and this will properly limit stats collection during that time, but if a node goes down unexpectedly it will also block stats collection and that may be very bad for keeping a live system from failing due to high load (since stats about its load will be stopped).

We also don't yet have logic in the deploy service to detect and remove/redeploy dead nodes, so any cluster that got into this state wouldn't get out (this latter point should be solved soon, but this change here will exacerbate it).

Distinguishing these two is going to be tough, so I won't block this for now, but I think we need to make an issue to track solving this problem. Perhaps a simple flag put into the DNS TXT record on "first time startup" or not, but there would also still need to be some logic there to handle "never actually started" so I don't think that's it, either.

Reply:
I'm reviewing the code now and the `stats` are only being collected by the leader for itself, right? Probably this check would make sense just ensuring that I'm the leader and collect stats if I'm up, does not matter if any other random node is down we do not affect this logic. What do you think?

Reply:
That's true, but also was intended to be changed in the future. We are relying on an assumption that `request volume >> load per request` that works for a lot of "webby" things, but not "hard compute" things, and we wanted to have the cluster also share their stats with the leader so a snapshot of the whole cluster is shoved up to the deploy service, but that hasn't been done yet.

Reply:
It might be the right choice for now, though, and we revisit when we have whole-cluster stats tracking?

Reply:
Even if the `get_stats` method collect the data from all up nodes and not only from the leader, we would still be only interested in call this method from the leader, so we will interested on do it when the leader is up, so the logic might be the same. But well that just be true once we revisit the logic as you said. Will make the change for now.